src/main/java/io/vertx/codegen/type/AnnotationValueTypeInfo.java

```diff
@@ -0,0 +1,257 @@
+package io.vertx.codegen.type;
+
+import java.util.List;
+
+/**
+ * @author <a href="mailto:cafeinoman@openaliasbox.org>Francois Delalleau</a>
+ */
+public class AnnotationValueTypeInfo {
```

Comment:
well type cast must be done, except that calling getShortValue() does the cast automatically but it still happen. I prefer to use a Map to limit the API surface and remain clear in the intent.